### ISSUE TITLE
Training time is too long.

### ISSUE THREAD

**Issue by tuntun-sd** [ybarancan/STSU#14]
Hello, thank you for your work. I have encountered some problems during training. When I trained with the default parameters, I found that it was very slow and it took about 20 days to train 200 epochs. After elimination, I found that the default batch_size was 1. So I set the batch_size to 4. But an error occurred during the run. Could you please tell me what batch_size you set during training?

![problem](https://user-images.githubusercontent.com/71703600/222383150-8349d0e4-c2af-43fc-a0e2-73a3c5003ea3.png)

**Comment by ybarancan** [comment]
Hi, The method assumes batch size of 1 for some steps. It shouldn't be difficult to extend it to higher batch size but I am not planning to do so in the near future. That epoch limit is not binding. What I do is monitor validation accuracy on calibration scenes and stop after reaching the peak which happens much earlier than 200 epochs.

**Comment by tuntun-sd** [comment]
I see, thank you for the explanation. It sounds like the method works well for smaller batch sizes and you have a way to stop training when accuracy reaches its peak. It's important to be efficient and avoid over-fitting, and it seems like you have found a good balance.

**Comment by Zhutianyi7230** [comment]
> 

Hi. I'm training this work with batch_size=1 and monitoring the loss and accuracy. 
Can you tell me the approximate loss and accuracy when you stop training, and how many epochs you trained in total before you stopped training?
Thanks a lot.
